packages/web-components/fast-foundation/src/data-grid/data-grid-cell.ts

```diff
@@ -0,0 +1,310 @@
+}
+
+/**
+ * A Data Grid Cell Custom HTML Element.
+ *
+ * @public
+ * @beta
+ */
+export class DataGridCell extends FASTElement {
+    /**
+     * The type of cell
+     *
+     * @public
+     * @remarks
+     * HTML Attribute: cell-type
+     */
+    @attr({ attribute: "cell-type" })
+    public cellType: DataGridCellTypes;
+    private cellTypeChanged(): void {
+        if ((this as FASTElement).$fastController.isConnected) {
+            this.updateCellView();
+        }
+    }
+
+    /**
+     * The column index of the cell.
+     * This will be applied to the css grid-column-index value
+     * applied to the cell
+     *
+     * @public
+     * @remarks
+     * HTML Attribute: grid-column
+     */
+    @attr({ attribute: "grid-column" })
+    public gridColumn: string;
+    private gridColumnChanged(): void {
+        if ((this as FASTElement).$fastController.isConnected) {
+            this.updateCellStyle();
+        }
+    }
+
+    /**
+     * The base data for the parent row
+     *
+     * @public
+     */
+    @observable
+    public rowData: object | null = null;
+    private rowDataChanged(): void {
+        if ((this as FASTElement).$fastController.isConnected) {
+        }
+    }
+
+    /**
+     * The base data for the column
+     *
+     * @public
+     */
+    @observable
+    public columnDefinition: ColumnDefinition | null = null;
+    private columnDefinitionChanged(
+        oldValue: ColumnDefinition | null,
+        newValue: ColumnDefinition | null
+    ): void {
+        if ((this as FASTElement).$fastController.isConnected) {
+            this.updateCellView();
+        }
+    }
+
+    private isActiveCell: boolean = false;
+    private customCellView: HTMLView | null = null;
+    private isInternalFocused: boolean = false;
+
+    /**
+     * @internal
+     */
+    public connectedCallback(): void {
+        super.connectedCallback();
+
+        this.addEventListener("focusin", this.handleFocusin);
+        this.addEventListener("focusout", this.handleFocusout);
+        this.addEventListener("keydown", this.handleKeydown);
+
+        this.style.gridColumn = `${
+            this.columnDefinition?.gridColumn === undefined
+                ? 0
+                : this.columnDefinition.gridColumn
+        }`;
+
+        this.updateCellView();
+        this.updateCellStyle();
+    }
+
+    /**
+     * @internal
+     */
+    public disconnectedCallback(): void {
+        super.disconnectedCallback();
+
+        this.removeEventListener("focusin", this.handleFocusin);
+        this.removeEventListener("focusout", this.handleFocusout);
+        this.removeEventListener("keydown", this.handleKeydown);
+
+        this.disconnectCellView();
+    }
+
+    public handleFocusin(e: FocusEvent): void {
+        if (this.isActiveCell || this.columnDefinition === null) {
+            return;
+        }
+
+        this.isActiveCell = true;
+
+        switch (this.cellType) {
+            case undefined:
```

Comment:
Is there a case where we *don't* want to handle focusin? If not, should this just use the `default` case?

Reply:
yep